### PR TITLE
Update to Weld 5.1.0.Final

### LINKED ISSUE #22803: Update to Weld 5.1.0.Final
Weld 5.1.0.Final is now available along with Weld API 5.0.SP3.
Probe is no longer part of these artifacts.
Release notes can be seen at http://weld.cdi-spec.org/news/2022/10/03/weld-510Final/ as usual.

### PR DESCRIPTION
**Pull request by tevans78**
fixes https://github.com/OpenLiberty/open-liberty/issues/22803

### CODE REVIEW

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

File: dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        assertStringsInLogs("PROBE-000002: Processing bean deployment archive: " + bda);
+    }
+
+    /**
+     * Post test processing.
+     *
+     * @throws Exception
+     */
+    @AfterClass
+    public static void cleanup() throws Exception {
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java]: Can we move cleanup() so that it's next to setup()?

File: dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    @Test
+    public void testExtensionLoaded() throws Exception {
+        assertStringsInLogs("WELD-000411: Observer method \\[BackedAnnotatedMethod\\] public org.jboss.weld.probe.ProbeExtension.processAnnotatedTypes");
+        assertStringsInLogs("Weld Development Mode: ENABLED");
+
+        assertProbeProcessedBDA("multipleWars2#org.jboss.weld.probe.ProbeExtension.additionalClasses");
+        assertProbeProcessedBDA("multipleWars2#multipleWar1.war");
+        assertProbeProcessedBDA("multipleWars2#multipleWar1.war#WEB_INF_LIB#WEB-INF/lib/multipleWarEmbeddedJar.jar");
+        //assertProbeProcessedBDA("multipleWars2#com.ibm.ws.beanvalidation.v11.cdi_1.additionalClasses"); //BDA name is very different in EE9
+        //assertProbeProcessedBDA("multipleWars2#com.ibm.ws.jsf.2.2_1.additionalClasses");                //BDA name is very different in EE9
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java]: Remove commented out code Could add a new comment that we don't check for the BDAs added by BV or JSF since the name varies by version. Alternatively you could assert that one of the expected BDAs is present.

File: dev/com.ibm.ws.cdi.extension_fat/publish/servers/weldDevModeServer/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/dropins
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.extension_fat/publish/servers/weldDevModeServer/.gitignore]: Why is this needed? We shouldn't be putting anything into `/dropins` at build time.

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java [hunk 4]
```diff
@@ -494,10 +494,10 @@ public Iterable<Metadata<Extension>> getExtensions() {
...
             //if the probe is enabled, add the probe extension
-
-            if (CDIUtils.isDevelopementMode()) {
-                extensionSet.add(getProbeExtension());
-                WebSphereBeanDeploymentArchive bda = getBeanDeploymentArchive(ProbeExtension.class);
+            WeldDevelopmentMode devMode = this.cdiRuntime.getWeldDevelopmentMode();
+            if (devMode != null) {
+                extensionSet.add(devMode.getProbeExtension());
+                WebSphereBeanDeploymentArchive bda = devMode.getProbeBDA(this);
                 extensionBDAs.put(bda.getId(), bda);
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java:501]: Why do we need to add the probe extension BDA in two places? (Both here and in CDIContainerImpl)

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIRuntimeImpl.java [hunk 5]
```diff
@@ -699,4 +703,15 @@ public BuildCompatibleExtensionFinder getBuildCompatibleExtensionFinder() {
...
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public WeldDevelopmentMode getWeldDevelopmentMode() {
+        WeldDevelopmentMode devMode = null;
+        //only return the service if it exists and is enabled
+        if (this.weldDevelopmentMode != null && this.weldDevelopmentMode.enabled()) {
+            devMode = this.weldDevelopmentMode;
+        }
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIRuntimeImpl.java]: Given that `this.weldDevelopmentMode` is volatile and so could change value while this code is running, I think we should copy it to a local variable before checking for `null` and enablement.

File: dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java (hunk 1 above)
Comment (tevans78) [reply dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java]: Done

File: dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java (hunk 2 above)
Comment (tevans78) [reply dev/com.ibm.ws.cdi.extension_fat/fat/src/com/ibm/ws/cdi/extension/tests/WeldDevModeTest.java]: Since we don't really need to check for ALL probe BDAs, it is easier to just ignore these ones

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java (hunk 4 above)
Comment (tevans78) [reply dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java:501]: The two places are performing different functions. The CDIContainerImpl code is creating the Probe BDA. This code is getting the Extension Metadata.

File: dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIRuntimeImpl.java (hunk 5 above)
Comment (tevans78) [reply dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIRuntimeImpl.java]: ok

File: dev/com.ibm.ws.cdi.extension_fat/publish/servers/weldDevModeServer/.gitignore (hunk 3 above)
Comment (tevans78) [reply dev/com.ibm.ws.cdi.extension_fat/publish/servers/weldDevModeServer/.gitignore]: removed

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
#build

**Review by Azquelt — APPROVED** [review]

**Comment by tevans78** [comment]
#libby
